.github/workflows/publish.yml

```diff
@@ -36,6 +37,22 @@ jobs:
           gh run cancel ${{ github.run_id }}
           gh run watch ${{ github.run_id }}
 
+      - name: Cancel all release preparation runs
+        run: |
+          # Get a list of all running or pending release preparation runs
+          # combining both the prepare-hotfix-release.yml and prepare-next-release.yml workflows
+          RUNNING_RELEASE_PREPARATIONS=$(
+            {
+              gh run list --limit 50 --workflow=prepare-hotfix-release.yml --json databaseId,status
+              gh run list --limit 50 --workflow=prepare-next-release.yml --json databaseId,status
+            } | jq -rc '.[] | select(.status | contains("in_progress", "pending", "queued", "requested", "waiting")) | .databaseId'
+          )
+
+          # Loop through each run and pass it to the "gh run cancel" command
+          while IFS= read -r databaseId; do
+            gh run cancel "$databaseId"
+          done <<< "$RUNNING_RELEASE_PREPARATIONS"
+
```

Comment:
this turned out to be a bit more complicated than I first thought (it works though). You could argue that this should become a js script instead that we can run unit tests against.